code/Pagetypes/FreshdeskPage.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+
+class FreshdeskPage extends Page
+{
+}
+
+class FreshdeskPage_Controller extends Page_Controller
+{
+    /**
+     * Priority mapping for Freshdesk tickets.
+     * @var array
+     */
+    private static $freshdeskPriority = [
+        0 => 'any',
+        1 => 'low',
+        2 => 'medium',
+        3 => 'high',
+        4 => 'urgent',
+    ];
+
+    private static $dependencies = [
+        'freshdesk' => '%$FreshdeskAPI',
+    ];
+
+    /**
+     * @var FreshdeskAPI
+     */
+    public $freshdesk;
+
+    /**
+     * @var array
+     */
+    private static $_tickets = [];
+
+    /**
+    * Returns Freshdesk tickets for a user
+    *
+    * @return PaginatedList
+    */
+    public function getTickets()
+    {
+        $currentMember = \Member::currentUser();
+        if (!$currentMember || !$currentMember->exists()) {
+            return \Security::permissionFailure();
+        }
+
+        if (!self::$_tickets) {
+            self::$_tickets = $this->freshdesk->getUserTickets($currentMember);
+        }
+
+        $tickets = self::$_tickets;
+
+        if (empty($tickets) || count($tickets) == 0) {
+            return false;
+        }
+
+        // Get all open by default
+        $filter = ['status' => 2];
+        if ($this->request->getVar('status')) {
+            $filter['status'] = (int) $this->request->getVar('status');
+        }
+        if ($this->request->getVar('priority')) {
+            $filter['priority'] = (int) $this->request->getVar('priority');
+        }
+
+        $tickets = $this->filterTickets($tickets, $filter);
+        $tickets = $this->freshdesk->humanReadable($tickets, self::$freshdeskPriority);
```

Comment:
There is some confusion here about who owns the concept of "human readable"? If it is the controller, then move humanReadable method into this class, and all the associated code. If it is not (which I recommend), then remove the humanReadable *call* from this class, and also rid of the `freshdeskPriority` - move it into FreshdeskAPI.

To look at it from yet another perspective: I'm suspecting the Controller here has no use for non-human values, so why not provide human already? Also, requiring consumer-developers of the FreshdeskAPI to call humanReadable is not good UX :-)